controllers/Backend.php

```diff
@@ -1,104 +1,351 @@
+                $errors['content'] = "Votre article doit contenir au moins 5 caractères.";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+  
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable.";
+                }
+            }
+            
+            if(empty($errors)){
+                $admin = $_SESSION['admin'];
+                $manager_article->edit($title,$chapo,$content,$posted, $admin, filter_input(INPUT_GET, 'id'));
+             
+                if(!empty($_FILES['image']['name']))
+                {
+                    $manager_article->update_img($_FILES['image']['tmp_name'], $extension);
+                    header("Location:/liste-de-tous-les-articles");
+                }
+                else
+                {
+                    header("Location:/liste-de-tous-les-articles");
+                }
+            }}else{
+                $errors = []; 
+            }
+        
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/postback";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if($posts == false){
+            header("Location:/page-erreur");
+        }
+
+        else{
+            $myView = new View('postback', 'backend');
+            $myView->render(array('post' => $posts, 'errors' => $errors, 'admins' => $admins));
+        }
     }
 
     public function login()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+    
+         if(filter_has_var(INPUT_POST, 'submit')){
+            if(filter_has_var(INPUT_POST, 'email')){
+                $email = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'email', FILTER_VALIDATE_EMAIL)));
+            }
+            if(filter_has_var(INPUT_POST, 'password')){
+                $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            }
+            
+            $errors = [];
+
+            if(empty($email) || empty($password)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis!";
+            }else if($manager_user->is_admin($email,$password) == 0){
+                $errors['exist']  = "Cet administrateur n'existe pas";
+            }
+                    
+            if(empty($errors)){
+                $_SESSION['admin'] = $email;
+                header("Location: /dashboard");
+
+            }
+        }
+        else
+            $errors =[];
         
-        $page="backend/login";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if(isset($_SESSION['admin'])){
+            header("location: /dashboard");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function header() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)